message_sender/tasks.py

```diff
@@ -77,6 +79,44 @@ def run(self, metric_name, metric_value, session=None, **kwargs):
 fire_metric = FireMetric()
 
 
+class Concurrency_Limiter(object):
+    def __init__(self):
+        self.redis_server = redis.StrictRedis(host=settings.REDIS_HOST,
+                                              port=settings.REDIS_PORT,
+                                              db=settings.REDIS_DB)
+
+    def get_current_message_count(self, msg_type):
+        keys = self.redis_server.keys(pattern=msg_type + "_messages_at_*")
+        total = 0
+        # Sum the values in all the buckets to get the total
+        for key in keys:
+            total += int(self.redis_server.get(key))
+        return total
+
+    def incr_message_count(self, msg_type, delay):
+        # Buckets of 1min
+        key = msg_type + "_messages_at_" + time.strftime("%M")
```

Comment:
We probably won't have this case, but by setting the key to the minute value, if we set the `delay` to longer than 1 hour, we're going to run into issues. A better way of doing this might be to set the key to `current_time_in_seconds // bucket_size_in_seconds`, so that we will never have the same bucket name for any two buckets.